core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/salt/rotatepassword/RotateSaltPasswordActions.java

```diff
@@ -0,0 +1,91 @@
+import com.sequenceiq.cloudbreak.core.flow2.cluster.salt.update.SaltUpdateService;
+import com.sequenceiq.cloudbreak.core.flow2.externaldatabase.StackUpdaterService;
+import com.sequenceiq.cloudbreak.core.flow2.stack.provision.action.AbstractStackCreationAction;
+import com.sequenceiq.cloudbreak.core.flow2.stack.start.StackCreationContext;
+import com.sequenceiq.cloudbreak.reactor.api.event.StackEvent;
+import com.sequenceiq.cloudbreak.reactor.api.event.cluster.RotateSaltPasswordFailureResponse;
+import com.sequenceiq.cloudbreak.reactor.api.event.cluster.RotateSaltPasswordRequest;
+import com.sequenceiq.cloudbreak.reactor.api.event.cluster.RotateSaltPasswordSuccessResponse;
+
+@Configuration
+public class RotateSaltPasswordActions {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RotateSaltPasswordActions.class);
+
+    @Inject
+    private SaltUpdateService saltUpdateService;
+
+    @Inject
+    private StackUpdaterService stackUpdaterService;
+
+    @Bean(name = "ROTATE_SALT_PASSWORD_STATE")
+    public Action<?, ?> rotateSaltPassword() {
+        return new AbstractStackCreationAction<>(StackEvent.class) {
+
+            @Override
+            protected void doExecute(StackCreationContext context, StackEvent payload, Map<Object, Object> variables) throws Exception {
+                saltUpdateService.rotateSaltPassword(context.getStack());
+                sendEvent(context);
+            }
+
+            @Override
+            protected Selectable createRequest(StackCreationContext context) {
+                return new RotateSaltPasswordRequest(context.getStack().getId());
+            }
+        };
+    }
+
+    @Bean(name = "ROTATE_SALT_PASSWORD_SUCCESS_STATE")
+    public Action<?, ?> rotateSaltPasswordFinished() {
+        return new AbstractStackCreationAction<>(RotateSaltPasswordSuccessResponse.class) {
+
+            @Override
+            protected void doExecute(StackCreationContext context, RotateSaltPasswordSuccessResponse payload, Map<Object, Object> variables) throws Exception {
+                stackUpdaterService.updateStatus(payload.getResourceId(), DetailedStackStatus.AVAILABLE,
+                        CLUSTER_SALT_PASSWORD_ROTATE_FINISHED, "Salt password rotated");
+                sendEvent(context);
+            }
+
+            @Override
+            protected Selectable createRequest(StackCreationContext context) {
+                return new StackEvent(RotateSaltPasswordEvent.ROTATE_SALT_PASSWORD_FINISHED_EVENT.event(), context.getStack().getId());
+            }
+        };
+    }
+
+    @Bean(name = "ROTATE_SALT_PASSWORD_FAILED_STATE")
+    public Action<?, ?> rotateSaltPasswordFailed() {
+        return new AbstractStackCreationAction<>(RotateSaltPasswordFailureResponse.class) {
+
+            @Override
+            protected void doExecute(StackCreationContext context, RotateSaltPasswordFailureResponse payload, Map<Object, Object> variables) throws Exception {
```

Comment:
I think a log with the error would be nice here. also the logger wouldn't be even used in this class